java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -34,105 +34,110 @@ public class CommandManager {
-                .setSingleCommand(
-                        RedisRequestOuterClass.Command.newBuilder()
-                                .setRequestType(mapRequestTypes(command))
-                                .setArgsArray(commandArgs.build())
-                                .build())
-                .setRoute(
-                        RedisRequestOuterClass.Routes.newBuilder()
-                                .setSimpleRoutes(RedisRequestOuterClass.SimpleRoutes.AllNodes)
-                                .build());
-    }
+        var builder =
+                RedisRequest.newBuilder()
+                        .setSingleCommand(
+                                Command.newBuilder()
+                                        .setRequestType(requestType)
+                                        .setArgsArray(commandArgs.build())
+                                        .build());
 
-    private RequestType mapRequestTypes(Command.RequestType inType) {
-        switch (inType) {
-            case CUSTOM_COMMAND:
-                return RequestType.CustomCommand;
-        }
-        throw new RuntimeException("Unsupported request type");
+        return prepareRedisRequestRoute(builder, Optional.of(route));
     }
 
     /**
-     * Build a protobuf command/transaction request object with routing options.<br>
-     * Used by {@link CommandManager}.
+     * Build a protobuf command request object.
      *
-     * @param command Redis command type
-     * @param args Redis command arguments
-     * @param route Command routing parameters
+     * @param requestType Redis command type
+     * @param arguments Redis command arguments
      * @return An uncompleted request. {@link CallbackDispatcher} is responsible to complete it by
      *     adding a callback id.
      */
-    private RedisRequest.Builder prepareRedisRequest(
-            Command.RequestType command, String[] args, Optional<Route> route) {
+    protected RedisRequest.Builder prepareRedisRequest(RequestType requestType, String[] arguments) {
         ArgsArray.Builder commandArgs = ArgsArray.newBuilder();
-        for (var arg : args) {
+        for (var arg : arguments) {
             commandArgs.addArgs(arg);
         }
 
         var builder =
                 RedisRequest.newBuilder()
                         .setSingleCommand(
-                                RedisRequestOuterClass.Command.newBuilder()
-                                        .setRequestType(mapRequestTypes(command))
+                                Command.newBuilder()
+                                        .setRequestType(requestType)
                                         .setArgsArray(commandArgs.build())
                                         .build());
 
+        return prepareRedisRequestRoute(builder, Optional.empty());
+    }
+
+    private RedisRequest.Builder prepareRedisRequestRoute(
+            RedisRequest.Builder builder, Optional<Route> route) {
         if (route.isEmpty()) {
```

Comment:
why take `route` as optional, instead of just not calling this function when there's no route? the functions calling this never take an optional route.